### ISSUE TITLE
Graph overview

### ISSUE THREAD

**Issue by d-ornen** [rizinorg/cutter#3001]
Processor: intel xeon e5-1650 v3
Dragging colored graph around consumes a lot of CPU and affects usability (lags, freezes).
![image](https://user-images.githubusercontent.com/46656839/180617956-2f7d1028-37ac-460f-9b21-e5a73c7f1938.png)
Feature request: optimize graph redraw system :pleading_face: :pray:

**Comment by karliss** [comment]
How big function are you looking at? How many basic blocks and edges? Is it important that all the basic blocks are colored? I assume you used some kind of script or plugin to automatically color all the blocks.

**Comment by d-ornen** [comment]
The function is  4069 bytes long
There are about 130 basic blocks and about 60 of them were colored.
I don't have such scripts i just have a lot of free time :)
However, i am thinking about implementing one.

Important note: The lags increase at zoom level where font begins to draw itself

**Comment by karliss** [comment]
Something doesn't seem right.  There is definitely room for improvements and the CPU usage is somewhat higher than ideal, but it should be nowhere near as bad to cause anything what  would consider lag and freezing especially with function of that size.   I did a quick test on a ~    6-7 year old laptop with 10 times bigger function and dragging the view around was sufficiently smooth. The delay when opening function and each time I changed color of a block was longer than I would like, but dragging the view around was good. It's only slightly newer than 8 year old server CPU you mentioned but by most performance metrics it should perform worse. Except maybe built-in GPU. Two more differences where that I tested on Linux and didn't spend time to color all the blocks.   Not having blocks colored should have little impact on performance, but the drawing backend differences between Linux and windows might be part of problem.    

You didn't answer to my previous question. Is coloring blocks important,   does it not lag while dragging if blocks aren't colored?

**Comment by d-ornen** [comment]
Yes, sorry. Colored blocks greatly reduce perfomance. About 30-40 blocks are enough to delay UI feedback for 7-10 seconds in worst cases.

**Comment by karliss** [comment]
Last question (maybe). Can you tell the information asked in issue template: the exact version of cutter you used and where you obtained it (downloaded binary from github, compiled yourself or something else)?

**Comment by karliss** [comment]
Tried on windows and highlighting ~30 blocks. Same laptop as before with intel 6500U.  Still couldn't repeat it. Didn't observe any significant lagging or freezing while dragging the view around.

**Comment by d-ornen** [comment]
```
Version 2.1.0
Using rizin 0.4.0 (7158b61)
Based on Qt 5.15.2 (MSVC 2017, 64 bit)
```
Obtained from releases page from [cutter-v2.1.0-Windows-x86_64.zip](https://github.com/rizinorg/cutter/releases/download/v2.1.0/cutter-v2.1.0-Windows-x86_64.zip) asset

**Comment by notxvilka** [comment]
@d-ornen please try again with the latest version
